### PR TITLE
Added api to pass errors from generators

### PR DESCRIPTION
**Pull request by KoryNunn**

### CODE REVIEW

File: test/index.js [hunk 1]
```diff
@@ -669,4 +669,30 @@ test('generators that yield promises :/', function(t){
...
+    var generated = righto.iterate(function*(reject){
+        var x = yield righto(function(done){
+            asyncify(function(){
+                done(null, 'x');
+            });
+        });
+
+        reject('foo');
+
+        var x = yield righto(function(done){
```
Comment (bguiz) [inline test/index.js]: `var y`

**Comment by KoryNunn** [comment]
I have some reservations about the PR straight away, but I'm not sure if they are valid.

calling `reject()` does not necessarily halt execution at that point, but rather, will allow execution to continue to the next `yield` or `return`. This could cause side effects, just as it would if you allowed an errback style function to call `callback(error)` without returning.

I could solve this by `throw`ing within the `reject` function, catching all throw errors, and detecting ones thrown by `reject`, swallowing, and passing the error, while re-throwing valid throws, but that would have a substantial impact on performance.

**Comment by bguiz** [comment]
I think it isn't worth the overhead. Also if you're using standard callbacks (without righto), you would already be accustomed to returning if the callback isn't at the last executable line of of the current function scope.

File: test/index.js [hunk 2]
```diff
@@ -669,4 +669,30 @@ test('generators that yield promises :/', function(t){
...
+    t.plan(1);
+
+    var generated = righto.iterate(function*(reject){
+        var x = yield righto(function(done){
+            asyncify(function(){
+                done(null, 'x');
+            });
+        });
+
+        reject('foo');
```
Comment (bguiz) [inline test/index.js]: perhaps change this to 

```
if (x === 'x') { 
  return reject('foo');  
}
```

**Comment by bguiz** [comment]
So see the diff comment above, so the test now demonstrates that a `return` is expected, just like you would if a callback had occurred at that point. 

Additional observation: Since `reject` takes the first argument as an error, perhaps we could just make it into a standard errback, so that 2nd arg is considered the result. So it could also be used to shortcut the iteration of the generator function. The `return` achieves this anyway, but would make for a more consistent experience with errbacks.

**Comment by KoryNunn** [comment]
I had coincided allowing to pass results there, but that does add ways to do the same task, which I'm not a huge fan of. It also wouldn't be a breaking change too add later, so, goat tracks and all that...

**Comment by bguiz** [comment]
> but that does add ways to do the same task

'nuff said... merge!

**Pull request merged**